### ISSUE TITLE
Property and method name clash since 2.3.7

### ISSUE THREAD

**Issue by nuernbergerA** [livewire/livewire#2387]
**Context**
- Livewire version: [v2.3.7](https://github.com/livewire/livewire/releases/tag/v2.3.7)
- Related PR #2352 

### Description
Property and method name clash since 2.3.7

### Exact steps to reproduce
**Component.php**
```php
class ShowPost extends Component
{
    public Post $post;

    public function post() {
        // do some work
    }
}
```

**post.blade.php**
```blade
<div>
    {{ $post->title }}

    {{-- or --}}

    @foreach($post->tags()->whereVisible()->get() as $tag)
        {{-- --}}
    @endforeach
</div>
```

Results in an error
`Call to undefined method Closure`

**Comment by ziming** [comment]
The livewire-select package had a similar error after updating to livewire 2.3.7 too.

**Comment by nuernbergerA** [comment]
### Current solution
**composer.json**
`"livewire/livewire": "2.3.6"`

**Comment by imliam** [comment]
This has also caused an issue on one of our apps that means we can't upgrade Livewire to the (at the time of writing) latest version. I'm also not a huge fan of the syntax of methods automatically being sent as invokable variables. I dislike this in Blade components too and wrote down a few reasons for that here: laravel/ideas#2129

**Comment by ryangjchandler** [comment]
@nuernbergerA This is closed by #2388 - Caleb reverted the PR mentioned in the thread.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)
